src/views/Rules/ListRules/index.jsx

```diff
@@ -254,7 +259,22 @@ function ListRules(props) {
       });
 
       setRulesWithScheduledChanges(sortedRules);
-    });
+      setRewoundRules([]);
+    }
+  }, [rules, scheduledChanges, requiredSignoffs]);
+
+  useEffect(() => {
+    fetchRules(rewindDate ? rewindDate.getTime() : null);
+  }, [rewindDate]);
```

Comment:
Do we `fetchRules` each time the `rewindDate` value change (i.e., a user clicking on different dates inside the widget)? This might explain why the picker is lagging.